### PR TITLE
avoid base64, remove unused filtering in server arguments

### PR DESCRIPTION
**Pull request by lasote**
- I think we should be ok sending the contents of the file. The json encoding/decoding should be able to escape all the needed quotes and weird chars.
- Removed double filtering in client and removed arguments to filter in the server (we were sending None).

### CODE REVIEW

File: conans/server/service/common/search.py [hunk 1]
```diff
@@ -61,7 +60,8 @@ def search_packages(server_store, ref, query, look_in_all_rrevs):
     if not os.path.exists(server_store.conan_revisions_root(ref.copy_clear_rev())):
         raise RecipeNotFoundException(ref)
     infos = _get_local_infos_min(server_store, ref, look_in_all_rrevs)
-    return filter_packages(query, infos)
+    assert query is None, "The server is not filtering packages remotely anymore"
```
Comment (memsharded) [inline conans/server/service/common/search.py:63]: Maybe add a version client message necessary, because this might crash for older Conan clients?

**Review by memsharded — APPROVED** [review]
Looking good, my first trials were failing, so I assumed it was because of json not scaping good enough the strings, but apparently it works

File: conans/server/service/common/search.py (hunk 1 above)
Comment (lasote) [reply conans/server/service/common/search.py:63]: But very very old. We can remove the assert if you want.

Comment (memsharded) [reply conans/server/service/common/search.py:63]: It is fine, lets leave it.

**Pull request merged**